### PR TITLE
Fix remove down node

### PR DESCRIPTION
**Pull request by jrochas**

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/ad10904845a89986217331b7161d551b1494b30f/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/ca4710a04f1d07ad109f0cde4ab590a16d2de362/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 1]
```diff
@@ -637,29 +637,30 @@ public void acquireAllNodes(Map<String, ?> nodeConfiguration) {
...
      */
     public BooleanWrapper removeNode(String nodeUrl, Client initiator) {
-
+        Node node = null;
         //verifying if node is already in the list,
         //node could have fallen between remove request and the confirm
         if (this.nodes.containsKey(nodeUrl)) {
-            logger.info("[" + name + "] removing node : " + nodeUrl);
-            Node node = nodes.remove(nodeUrl);
+            logger.info("Remove node " + nodeUrl + " from node source " + this.name);
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: I think it is better to keep the node source logging semantic [nodesourceName] etc

Comment (jrochas) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: Ok no problem, I reverted it to the previous log format. Thanks.

**Comment by jrochas** [comment]
retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/61abca53be16609f767cc869296908e9749a57e3/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 2]
```diff
@@ -637,29 +637,29 @@ public void acquireAllNodes(Map<String, ?> nodeConfiguration) {
...
+            node = this.nodes.remove(nodeUrl);
+        } else if (this.downNodes.containsKey(nodeUrl)) {
+            logger.info("[" + this.name + "] removing down node: " + nodeUrl);
+            node = this.downNodes.remove(nodeUrl);
+        }
+
+        if (node == null) {
+            logger.error("[" + this.name + "] cannot remove node: " + nodeUrl + " because it is unknown");
+            return new BooleanWrapper(false);
+        } else {
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: small side effect is that a warning message will be printed by TopologyManager when removing a down node. Not a big deal.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [NodeSource.java#L566](https://github.com/ow2-proactive/scheduling/blob/10a2bd91078dcf7321bbda12fb47b5357ed8df0f/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java#L566): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**